### PR TITLE
Make Card explicitChildNodes vs container be configurable

### LINKED ISSUE #14248: a11y: card should not be focusable as a whole if it contains sub-elements
From a recent discussion:

"Don't focus the entire card as shown in the screenshot. A card usually contains a single entity of information that doesn't require the user to navigate or understand groupings at a more granular level. Since your UI will require this, focusing the card can actually cause focus to move around in unexpected ways when using "Explore by touch". Instead, ensure focus moves to elements within the card."

### PR DESCRIPTION
**Pull request by jonahwilliams**
Allow configuring whether a card should have explicit semantic nodes or be a container

Fixes https://github.com/flutter/flutter/issues/14248

### CODE REVIEW

File: packages/flutter/test/material/card_test.dart [hunk 1]
```diff
@@ -39,22 +39,25 @@ void main() {
...
-                  SemanticsAction.tap,
-                ],
-                flags: <SemanticsFlag>[
-                  SemanticsFlag.isButton,
-                  SemanticsFlag.hasEnabledState,
-                  SemanticsFlag.isEnabled,
-                ],
-              ),
+          ),
+          new TestSemantics.rootChild(
```
Comment (Hixie) [inline packages/flutter/test/material/card_test.dart]: multiple `rootChild`s ?

File: packages/flutter/test/material/card_test.dart [hunk 2]
```diff
@@ -39,22 +39,25 @@ void main() {
...
+            id: 3,
+            label: 'Button',
+            textDirection: TextDirection.ltr,
+            actions: <SemanticsAction>[
+              SemanticsAction.tap,
+            ],
+            flags: <SemanticsFlag>[
+              SemanticsFlag.isButton,
+              SemanticsFlag.hasEnabledState,
+              SemanticsFlag.isEnabled,
```
Comment (Hixie) [inline packages/flutter/test/material/card_test.dart]: Can you add a test that verifies that if the card just contains text, it still ends up being only a single node?

File: packages/flutter/test/material/card_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter/test/material/card_test.dart]: Done

File: packages/flutter/test/material/card_test.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter/test/material/card_test.dart]: Hm, I guess this won't really work. I'm not sure we have a way to express this in our current system.

**Comment by jonahwilliams** [comment]
PTAL, based on offline discussion with @goderbauer we decided that we don't really have a good way to automagically merge semantics.  Instead, we can make it configurable to ~~pass the problem onto the developer~~ let the developer choose the correct granularity.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**